### PR TITLE
Better new beatmap discussions footer alignment

### LINKED ISSUE #9228: Spelling error/responsiveness in beatmap discussion pages (Hype)
Needs a space between "hype" and "in" as per this screenshot: 
image

Also might want to check the responsiveness on mobile:
image

### PR DESCRIPTION
**Pull request by notbakaneko**
closes #9228

React and flexbox fun 🤨

### CODE REVIEW

File: resources/assets/lib/beatmap-discussions/new-discussion.tsx [hunk 1]
```diff
@@ -356,16 +356,19 @@ export class NewDiscussion extends React.Component<Props> {
...
-                      }}
-                      pattern={` ${osu.trans('beatmaps.hype.new_time')}`}
-                    />
+                    <>
+                      {' '}
+                      <StringWithComponent
+                        mappings={{
+                          new_time: <TimeWithTooltip dateTime={this.props.beatmapset.current_user_attributes.new_hype_time} relative />,
+                        }}
+                        pattern={` ${osu.trans('beatmaps.hype.new_time')}`}
```
Comment (nanaya) [inline resources/assets/lib/beatmap-discussions/new-discussion.tsx:365]: one of the extra spaces isn't needed

File: resources/assets/less/bem/beatmap-discussion-new.less [hunk 2]
```diff
@@ -23,21 +23,18 @@
 
   &__footer {
     display: flex;
-    flex-wrap: wrap;
```
Comment (nanaya) [inline resources/assets/less/bem/beatmap-discussion-new.less:26]: The buttons look rather wrong. The box message could also use some padding (this one is an existing problem though).

File: resources/assets/less/bem/beatmap-discussion-new.less [hunk 3]
```diff
@@ -23,21 +23,18 @@
...
+    gap: 5px;
     background-color: @osu-colour-b5;
-    padding: 5px @_side-padding;
+    padding: 10px @_side-padding;
     justify-content: space-between;
 
     @media @mobile {
-      width: 100%;
-      flex-direction: column;
```
Comment (nanaya) [inline resources/assets/less/bem/beatmap-discussion-new.less:33]: Button overflow due to this change

**Review by nanaya — CHANGES_REQUESTED** [review]

File: resources/assets/less/bem/beatmap-discussion-new.less (hunk 2 above)
Comment (notbakaneko) [reply resources/assets/less/bem/beatmap-discussion-new.less:26]: More like text is too long and should be wrapping anyway.

I intentionally changed this so the timestamp message would not shift the buttons:
image
image

Either the text should be wrapping or the elements should always be stacking (or buttons should not grow which then causes elements to not align visually)

Comment (nanaya) [reply resources/assets/less/bem/beatmap-discussion-new.less:26]: I don't think the tall button is a good compromise

File: resources/assets/less/bem/beatmap-discussion-new.less [hunk 4]
```diff
@@ -49,6 +46,23 @@
     }
   }
 
+  &__footer-column {
```
Comment (nanaya) [inline resources/assets/less/bem/beatmap-discussion-new.less:49]: the commit message says less confusing name but there are two footer columns and this only for one specific column, not both 🤨

File: resources/assets/less/bem/beatmap-discussion-new.less (hunk 2 above)
Comment (peppy) [reply resources/assets/less/bem/beatmap-discussion-new.less:26]: Would also prefer the buttons to not get taller.

**Review by nanaya — APPROVED** [review]

**Pull request merged**